src/librustc_errors/emitter.rs

```diff
@@ -115,6 +116,33 @@ struct FileWithAnnotatedLines {
     multiline_depth: usize,
 }
 
+impl Drop for EmitterWriter {
+    fn drop(&mut self) {
```

Comment:
@GuillaumeGomez (more belated commentary in light of #48550) What was the motivation for putting the this in `drop()`? Semantically, printing the message notifying the user about `--explain` does not seem like a "cleanup" action. As [Niko suggested](https://github.com/rust-lang/rust/issues/48041#issuecomment-363842863), I would kind of expect this to go with [the "aborting due to previous error" message in `Handler.abort_if_errors`](https://github.com/rust-lang/rust/blob/89e5a0796e0c5de52cd1d6c7541e5c2c7c47f2cd/src/librustc_errors/lib.rs#L562-L564)—and it looks like [the `Handler` already knows which codes have been emitted](https://github.com/rust-lang/rust/blob/89e5a0796e0c5de52cd1d6c7541e5c2c7c47f2cd/src/librustc_errors/lib.rs#L604-L610), too.

Reply:
Hum, if this is true, it might allow to make my task for #48562.